### PR TITLE
Merge `Grouping` classes from Refractored.MvvmHelpers

### PR DESCRIPTION
**Pull request by tranb3r**
### Description of Change ###

Merge the `Grouping` classes from Refractored.MvvmHelpers (from James Montemagno).

### Bugs Fixed ###

Issue https://github.com/xamarin/XamarinCommunityToolkit/issues/786

### API Changes ###

Added:
- `class Grouping<TKey, TItem> : ObservableRangeCollection<TItem>`
- `class Grouping<TKey, TSubKey, TItem> : ObservableRangeCollection<TItem>`

### Behavioral Changes ###

No behavioral change, only additions.

### PR Checklist ###

- [x] Has tests
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: src/CommunityToolkit/Xamarin.CommunityToolkit.UnitTests/ObjectModel/Grouping_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	{
+		readonly Person[] people;
+
+		public Grouping_Tests()
+		{
+			people = new[]
+			{
+				new Person { FirstName = "Joseph", LastName = "Hill" },
+				new Person { FirstName = "James", LastName = "Montemagno" },
+				new Person { FirstName = "Pierce", LastName = "Boggan" },
```
Comment (jfversluis) [inline src/CommunityToolkit/Xamarin.CommunityToolkit.UnitTests/ObjectModel/Grouping_Tests.cs:17]: <3

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
Awesome, thank you! One thing, can you target this to develop? :)

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
I did it for you :)

**Comment by tranb3r** [comment]
> I did it for you :)

Thank you!

**Pull request merged**